02-express-tutorial/app.js

```diff
@@ -1 +1,50 @@
-console.log('Express Tutorial')
+const express = require("express");
+const app = express();
+const { products } = require("./data");
+
+app.use(express.static("./public"));
+
+app.get("/api/v1/test", (req, res) => {
+  res.json({ message: "It worked" });
+});
+
+app.get("/api/v1/products", (req, res) => {
+  res.send(products);
+});
+
+app.get("/api/v1/products/:productID", (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find((p) => p.id === idToFind);
+  if (!product) {
+    return res.status(404).json({ message: "That product was not found" });
+  }
+  res.json(product);
+});
+
+app.get("/api/v1/query", (req, res) => {
+  console.log(req.query);
+  let newProducts = [...products];
+  const { search, limit, priceLimit } = req.query;
+  if (search) {
+    newProducts = newProducts.filter((product) => {
+      return product.name.includes(search);
```

Comment:
The assignment says to have the search query be a "starts with" search rather than an "includes" search.  Functionally, your code is a better search solution, but it's always important to pay close attention to the requirements given!

Reply:
Thank you!  Yes, I had initially utilized the "starts with" query but decided to change it since step 13 just indicated to add more logic of our choosing :-)